tests/test_database.py

```diff
@@ -38,41 +79,112 @@ def setUpClass(self):
             self.test_database_connector)
         self.test_tag_manager = database_manager.TagManager(
             self.test_database_connector)
+        self.test_search_manager = search_manager.SearchManager(
+            self.test_tag_manager, self.test_item_manager)
 
     def setUp(self):
         create_test_database_setup(self.test_database_connector)
 
+    def test_get_id(self):
+        # arrange & act
+        id = database_helper.get_id()
+
+        # assert
+        self.assertEqual(type(id), str)
+        self.assertEqual(len(id), 36)
+
     def test_add_item(self):
-        count = self.test_database_connector.execute(
-            "SELECT COUNT(*) FROM Items;")
-        self.assertEqual(count['COUNT(*)'], 0)
-        self.test_item_manager.add_item()
-        count = self.test_database_connector.execute(
-            "SELECT COUNT(*) FROM Items;")
-        self.assertEqual(count['COUNT(*)'], 1)
+        # arrange
+        item_id = database_helper.get_id()
+
+        # act
+        self.test_item_manager.add_item(item_id)
+
+        # assert
+        self.assertEqual(count_entries(
+            self.test_database_connector, 'Items'), 1)
 
     def test_delete_item(self):
-        count = self.test_database_connector.execute(
-            "SELECT COUNT(*) FROM Items;")
-        self.assertEqual(count['COUNT(*)'], 0)
-        item_id = self.test_item_manager.add_item()
-        count = self.test_database_connector.execute(
-            "SELECT COUNT(*) FROM Items;")
-        self.assertEqual(count['COUNT(*)'], 1)
-        self.test_item_manager.delete_item(item_id)
-        count = self.test_database_connector.execute(
-            "SELECT COUNT(*) FROM Items;")
-        self.assertEqual(count['COUNT(*)'], 0)
+        # arrange
+        item_ids = test_items_tags_setup(
+            self.test_item_manager, self.test_tag_manager)
+
+        # act
+        self.test_item_manager.delete_item(item_ids[0])
+
+        # assert
+        self.assertEqual(count_entries(
+            self.test_database_connector, 'Items'), 3)
```

Comment:
As written before, `self.assertItemsCount(3)` and `self.assertTagsCount(3)` would be nice to have
and also perhaps even `self.assertTagexists("my_tag")`

Up to you, might be over the top.

Reply:
updated the helper functions I had to match this a bit more